pkg/tfbridge/typechecker/typechecker.go

```diff
@@ -161,7 +161,15 @@ func (v *TypeChecker) validatePropertyValue(
 
 	switch typeSpec.Type {
 	case "boolean":
-		if !propertyValue.IsBool() {
+		// Check for strings that are values "true" or "false".
```

Comment:
Let's try to avoid leaving confusing comments for future maintainers in the source. 

"These are handled as booleans in the bridge, so they should be skipped by the type checker."

This is not right. https://github.com/pulumi/pulumi-terraform-bridge/issues/2520 is the actual root cause that trips up the type-checker, and it would be great to have a TODO comment linking the issue in here and explaining the workaround.